geyser-plugin-manager/src/geyser_plugin_manager.rs

```diff
@@ -29,21 +38,115 @@ impl GeyserPluginManager {
         libpath: &str,
         config_file: &str,
     ) -> Result<(), Box<dyn Error>> {
-        type PluginConstructor = unsafe fn() -> *mut dyn GeyserPlugin;
+        // Load the dynamic library and reconstruct the Box<dyn GeyserPlugin>
         let lib = Library::new(libpath)?;
         let constructor: Symbol<PluginConstructor> = lib.get(b"_create_plugin")?;
         let plugin_raw = constructor();
-        let mut plugin = Box::from_raw(plugin_raw);
+        let mut plugin: Box<dyn GeyserPlugin> = Box::from_raw(plugin_raw);
+
+        // Run on_load to do necessary initializations
         plugin.on_load(config_file)?;
-        self.plugins.push(plugin);
-        self.libs.push(lib);
+
+        // Wrap in Arc<RwLock<_>>
+        let plugin = Arc::new(RwLock::new(plugin));
+        let lib = Arc::new(RwLock::new(lib));
+
+        self.plugins.push(Arc::clone(&plugin));
+        self.libs.push(Arc::clone(&lib));
+
+        let owned_path = std::path::PathBuf::from(libpath);
+        let owned_config = config_file.to_owned();
+        let handle = std::thread::spawn(move || {
```

Comment:
Spawn a thread for each plugin is an overkill. I think we do not want to do this busily. This can be better handled by an administrative RPC call, in the RPC the user can direct reload or load a new plugin. So that it can solve both use cases.

Reply:
administrative RPC sounds like a much better solution.